---
layout: post
title:  "Python包内文件相互引用Error"
date: 2019-02-14 17:09:15
categories: [编程语言,python]
tags: [python]
excerpt_separator: <!--more-->
---
Python包内文件相互引用Error
ModuleNotFoundError: No module named ''
<!--more-->

错误截图：

![err-pic](/images/微信截图_20190214171114.png)

错误代码：

![err-code-pic](/images/微信截图_20190214171150.png)

修正代码：

![fixed-code-pic](/images/微信截图_20190214171210.png)

原因：
python的sys.path不包含package1，package1内部的模块无法发现package1本身的模块。
所以即使是包本身用自己也需要追加包名。
